### PR TITLE
fix(no-unused-emit-declarations): detect emits in templates

### PR DESCRIPTION
**Pull request by waynzh**
try to fix https://github.com/vuejs/eslint-plugin-vue/issues/2337

### CODE REVIEW

File: tests/lib/rules/no-unused-emit-declarations.js [hunk 1]
```diff
@@ -98,6 +98,16 @@ tester.run('no-unused-emit-declarations', rule, {
...
     },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <button @click="emits('bar')">Bar</button>
+      </template>
+      <script setup>
+        const emits = defineEmits(['bar'])
+      `
```
Comment (FloEdelmann) [inline tests/lib/rules/no-unused-emit-declarations.js:109]:
```suggestion
      <script setup>
        const emits = defineEmits(['bar'])
      </script>
      `
```

File: tests/lib/rules/no-unused-emit-declarations.js [hunk 2]
```diff
@@ -453,6 +463,24 @@ tester.run('no-unused-emit-declarations', rule, {
...
     },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <button @click="emit('bar')">Bar</button>
+      </template>
+      <script setup>
+        const emit = defineEmits(['foo', 'bar'])
+      `,
```
Comment (FloEdelmann) [inline tests/lib/rules/no-unused-emit-declarations.js:474]:
```suggestion
      <script setup>
        const emit = defineEmits(['foo', 'bar'])
      </script>
      `,
```

**Review by FloEdelmann — APPROVED** [review]
Thanks for the fix! I have just two minor nits, otherwise this looks good to me :slightly_smiling_face:

File: lib/rules/no-unused-emit-declarations.js [hunk 3]
```diff
@@ -56,6 +56,17 @@ function hasReferenceId(value, setupContext) {
   )
 }
 
+/**
+ * Check if the given name matches emitReferenceIds variable name
+ * @param {string} name
+ * @param {Set<Identifier>} emitReferenceIds
+ * @returns {boolean}
+ */
+function verifyEmitVariableName(name, emitReferenceIds) {
```
Comment (FloEdelmann) [inline lib/rules/no-unused-emit-declarations.js:65]: I think we should rename this function:

```suggestion
function isEmitVariableName(name, emitReferenceIds) {
```

"verify" sounds like there is a "correct" and a "wrong" way.

File: lib/rules/no-unused-emit-declarations.js [hunk 4]
```diff
@@ -191,6 +202,15 @@ module.exports = {
           }
 
           const { contextReferenceIds, emitReferenceIds } = setupContext
+
+          // verify defineEmits variable in template
+          if (
+            callee.type === 'Identifier' &&
+            verifyEmitVariableName(callee.name, emitReferenceIds)
```
Comment (FloEdelmann) [inline lib/rules/no-unused-emit-declarations.js:209]:
```suggestion
            isEmitVariableName(callee.name, emitReferenceIds)
```

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you! Sorry for the late review.
I didn't realize that usage was not supported 😅

**Pull request merged**